src/forms/input/Input.tsx

```diff
@@ -61,46 +63,58 @@ export class Input extends React.PureComponent<InputProps> {
             size,
             type,
             children,
+            name,
         } = this.props;
         let classNames = ["clr-control-container", className];
+        const setId = this.props.id;
         if (disabled) classNames.push("clr-form-control-disabled");
         return isBoxed ? (
-            <div className="form-group">
-                <label>{label}</label>
-                <input
-                    type={type || "text"}
-                    id={name}
-                    value={value}
-                    defaultValue={defaultValue}
-                    size={size}
-                    disabled={disabled}
-                    placeholder={placeholder}
-                    onChange={this.handleChange}
-                    className={className}
-                />
-                {children}
-            </div>
-        ) : (
-            <div className="clr-form-control">
-                {label && Input.renderLabel(label)}
-                <div className={utils.classNames(classNames)}>
-                    <div className="clr-input-wrapper">
+            <UID>
+                {id => (
+                    <div className="form-group">
+                        <label>{label}</label>
                         <input
                             type={type || "text"}
-                            id={name}
+                            id={setId ? setId : id}
+                            name={name}
                             value={value}
                             defaultValue={defaultValue}
                             size={size}
                             disabled={disabled}
-                            className="clr-input"
                             placeholder={placeholder}
                             onChange={this.handleChange}
+                            className={className}
```

Comment:
I think we also need to keep class name "clr-input" and then add user define class names after that.

Reply:
I have not changes .... so dropping the comment